Greivance page

Aadeesh11 opened the pull request:
@devansh12b2.
Co-Authored-By: Devansh Mishra <79010144+devansh12b2@users.noreply.github.com>

Changes Made: Added Grievance page as per designs sent.
Page can be accessed from AppDrawer.

This PR will be ready for merging when we get the api info to fetch the required data from the backend. 
For now placeholder/hardcoded text/values have been used to match the design team's requirements.

Screenshots:
<img width="358" alt="Screenshot 2021-12-07 at 1 02 54 AM" src="https://user-images.githubusercontent.com/66922161/144910718-9f62a515-8d7c-4b57-abd8-09d2d0e95d5e.png"><img width="361" alt="Screenshot 2021-12-07 at 1 07 09 AM" src="https://user-images.githubusercontent.com/66922161/144910879-b3dae3bb-7364-4a7b-b306-23e906ac986b.png"><img width="361" alt="Screenshot 2021-12-07 at 1 03 01 AM" src="https://user-images.githubusercontent.com/66922161/144910743-40560abb-5720-4e6a-9a62-2da31d0cf593.png"><img width="360" alt="Screenshot 2021-12-07 at 1 03 06 AM" src="https://user-images.githubusercontent.com/66922161/144910772-80eacdb3-3b2e-4778-883d-9426f8c85f41.png">

devansh12b2 committed:
greivance folder created

Aadeesh11 committed:
counter card.

devansh12b2 committed:
chnages

devansh12b2 committed:
Merge branch 'Greivance_PAGE' of https://github.com/Aadeesh11/IIT-BHU-app into Greivance_PAGE

devansh12b2 committed:
chnages 2

Aadeesh11 committed:
Changes.

devansh12b2 committed:
text field

devansh12b2 committed:
Merge branch 'Greivance_PAGE' of https://github.com/Aadeesh11/IIT-BHU-app into Greivance_PAGE

Aadeesh11 committed:
Changes

devansh12b2 committed:
2nd

Aadeesh11 committed:
Revert "2nd"

This reverts commit 3a40f30cc04ba64c2778157d92d1be17b37c612c.

Aadeesh11 committed:
TextField

devansh12b2 committed:
dropdown

Aadeesh11 committed:
dropDownItems

devansh12b2 committed:
dropdown completed

Aadeesh11 committed:
styledDropDownButtons

devansh12b2 committed:
text fields

Aadeesh11 committed:
Dialogue UI.

Aadeesh11 committed:
ShowDialoue linked to confirm button

Aadeesh11 committed:
MORE Changes

devansh12b2 committed:
Color Scheme

Aadeesh11 committed:
Back icon changed to the required one

Aadeesh11 committed:
added all branches and limited the size of dropDOWN,

devansh12b2 committed:
Fonts added

devansh12b2 committed:
chngs

devansh12b2 committed:
font

Aadeesh11 committed:
Update greivance.dart

Co-Authored-By: Devansh Mishra <79010144+devansh12b2@users.noreply.github.com>

Vikhyath08 commented:
Some minor changes to be made:

- On entering text in "Describe your concern", the first line of the text overlaps the heading. Change this to give a little more space.
- Also, the green dot on the profile pic isn't necessary yet, so comment that out for now.
- Add a refresh indicator to the page, whose functionality will be implemented once the backend part is completed.
- Have a look at how workshop creation is done by looking at [workshopCreator.dart](https://github.com/IIT-BHU-InstiApp/IIT-BHU-app/blob/master/lib/model/workshopCreator.dart). Grievances will be created in a similar manner on clicking submit.
- As an extra feature, try making it so that even if I exit the page, the values which I had entered in are saved in the form.

devansh12b2 committed:
required changes

devansh12b2 committed:
required changes done

devansh12b2 committed:
fixes

devansh12b2 committed:
Merge branch 'master' into Greivance_PAGE

Vikhyath08 commented:
For the refreshing, use the RefreshIndicator widget(like [here](https://github.com/IIT-BHU-InstiApp/IIT-BHU-app/blob/82f66d236870649597b6cef4ed3fd5d32090f08c/lib/pages/club_entity/entityPage.dart#L221)).
Also, make sure that while submitting, none of the fields can be left empty apart from the drive link.
Apart from this, your PR looks good and is ready to merge!

Vikhyath08 commented:
Also, ~Greivance~ -> Grievance

devansh12b2 committed:
Refresh Indicator added

Vikhyath08 committed:
Removed Refresh from AppBar

Vikhyath08 committed:
Added validation

Aadeesh11 committed:
Connected Backend For Fetching Counters.

Co-Authored-By: Devansh Mishra <79010144+devansh12b2@users.noreply.github.com>

devansh12b2 committed:
bug fixes

Aadeesh11 committed:
Create Grievance functionality added.

1) 11 branches only.
2) course is B.Tech for all, since it is not taken as input. Wont need to pass course once backend is updated.

Co-Authored-By: Devansh Mishra <79010144+devansh12b2@users.noreply.github.com>

Aadeesh11 committed:
Validator for Drive link.

devansh12b2 committed:
ui fixes

devansh12b2 committed:
ui fixes

devansh12b2 committed:
ui fix

devansh12b2 committed:
print statements removed

Vikhyath08 commented:
Well done!

Just a minor change:
Add a snackbar to the Grievance Counts which, on clicking, says that this feature is under development, like [here](https://github.com/IIT-BHU-InstiApp/IIT-BHU-app/blob/82f66d236870649597b6cef4ed3fd5d32090f08c/lib/pages/academics/branchPage.dart#L38).

Aadeesh11 committed:
snackbar added.

Aadeesh11 committed:
getButton Function.

devansh12b2 committed:
Clear Button added

Aadeesh11 committed:
Updated Padding for snackbar

Aadeesh11 committed:
Fix: Grievance UI.

Aadeesh11 committed:
Updated error state while submitting greivance.
